test/Swashbuckle.AspNetCore.SwaggerGen.Test/SwaggerGenerator/SwaggerGeneratorTests.cs

```diff
@@ -845,7 +851,53 @@ public void GetSwagger_SupportsOption_DocumentFilters()
 
             var document = subject.GetSwagger("v1");
 
-            Assert.NotEmpty(document.Extensions);
+            Assert.Equal(2, document.Extensions.Count);
+            Assert.Equal("bar", ((OpenApiString)document.Extensions["X-foo"]).Value);
+            Assert.Equal("v1", ((OpenApiString)document.Extensions["X-docName"]).Value);
+        }
+
+        [Fact]
+        public void GetSwagger_SupportsOption_SchemaFilters()
+        {
+            var subject = new SwaggerGenerator(
+                new SwaggerGeneratorOptions
+                {
+                    SwaggerDocs = new Dictionary<string, OpenApiInfo>
+                    {
+                        ["v1"] = new OpenApiInfo { Version = "V1", Title = "Test API" }
+                    }
+                },
+                new FakeApiDescriptionGroupCollectionProvider(new[]
+                {
+                    ApiDescriptionFactory.Create<FakeController>(
+                        c =>
+                            nameof(c.ActionWithObjectParameter),
+                            groupName: "v1",
+                            httpMethod: "POST",
+                            relativePath: "resource",
+                            parameterDescriptions: new []
+                            {
+                                new ApiParameterDescription
+                                {
+                                    Name = "param",
+                                    Source = BindingSource.Body
+                                }
+                            })
+                }),
+                new SchemaGenerator(new SchemaGeneratorOptions
+                {
+                    SchemaFilters = new List<ISchemaFilter>
+                    {
+                        new VendorExtensionsSchemaFilter()
+                    }
+                }, new SystemTextJsonBehavior(new JsonSerializerOptions()))
+            );
+
+            var document = subject.GetSwagger("v1");
+
+            Assert.Equal(2, document.Components.Schemas["XmlAnnotatedType"].Extensions.Count);
+            Assert.Equal("bar", ((OpenApiString)document.Components.Schemas["XmlAnnotatedType"].Extensions["X-foo"]).Value);
+            Assert.Equal("v1", ((OpenApiString)document.Components.Schemas["XmlAnnotatedType"].Extensions["X-docName"]).Value);
```

Comment:
This should go in the `SystemTextJsonSchemaGeneratorTests` class. But that's OK, I can move it to thr right place rather than more back n forth.

Reply:
Looks good thx 👍